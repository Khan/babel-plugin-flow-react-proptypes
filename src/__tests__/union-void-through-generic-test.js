const babel = require('babel-core');
const content = `
var React = require('react');

type Maybe<T> = T |Â void;

type Props = {
  foo: Maybe<string>,
};

function C(props: Props) {
  return <div>{props.foo}</div>;
}
`;

it('union-void-through-generic', () => {
  const res = babel.transform(content, {
    babelrc: false,
    presets: ['@babel/env', '@babel/react', '@babel/flow'],
    plugins: [
      '@babel/syntax-flow',
      require('../'),
      "@babel/plugin-proposal-class-properties"
    ],
  }).code;
  expect(res).toMatchSnapshot();
});
