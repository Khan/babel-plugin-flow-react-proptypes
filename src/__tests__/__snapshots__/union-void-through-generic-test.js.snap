// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`union-void-through-generic 1`] = `
"\\"use strict\\";

var _propTypes = _interopRequireDefault(require(\\"prop-types\\"));

function _interopRequireDefault(obj) { return obj && obj.__esModule ? obj : { default: obj }; }

var React = require('react');

function C(props) {
  return React.createElement(\\"div\\", null, props.foo);
}

C.propTypes = {
  foo: _propTypes.default.string
};"
`;
